packages/headless/src/app/engine.ts

```diff
@@ -177,22 +177,22 @@ export function buildEngine<
   const engine = buildCoreEngine(options, thunkExtraArguments);
   const {accessToken, organizationId} = options.configuration;
   const {organizationEndpoints} = options.configuration;
-  let {platformUrl} = options.configuration;
-
-  if (shouldWarnAboutOrganizationEndpoints(options)) {
-    // @v3 make organizationEndpoints the default.
-    engine.logger.warn(
-      'The `organizationEndpoints` options was not explicitly set in the Headless engine configuration. Coveo recommends setting this option, as it has resiliency benefits and simplifies the overall configuration for multi-region deployments. See [Organization endpoints](https://docs.coveo.com/en/mcc80216).'
-    );
-  }
+  const platformUrl = organizationEndpoints?.platform
```

Comment:
refactored platform url init to one place with org end point overriding the platform url